sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/common/CounterSet.java

```diff
@@ -125,16 +125,16 @@ public synchronized boolean add(Counter<?> e) {
     if (null == e) {
       return false;
     }
-    if (counters.containsKey(e.getName())) {
+    if (counters.containsKey(e.getFlatName())) {
```

Comment:
counter equals is different than the counter name equals.
If two counters have the same name but different values, they are not equal.